macros/src/buffer.rs

```diff
@@ -208,19 +208,17 @@ fn impl_buffer_map_layout(
 
     Ok(quote! {
         impl #impl_generics ::bevy_impulse::BufferMapLayout for #struct_ident #ty_generics #where_clause {
-            fn buffer_list(&self) -> ::smallvec::SmallVec<[AnyBuffer; 8]> {
-                use smallvec::smallvec;
-                smallvec![#(
-                    self.#field_ident.as_any_buffer(),
-                )*]
-            }
-
             fn try_from_buffer_map(buffers: &::bevy_impulse::BufferMap) -> Result<Self, ::bevy_impulse::IncompatibleLayout> {
                 let mut compatibility = ::bevy_impulse::IncompatibleLayout::default();
                 #(
-                    let #field_ident = if let Ok(buffer) = compatibility.require_buffer_type::<#buffer>(#map_key, buffers) {
-                        buffer
-                    } else {
+                    let #field_ident = compatibility.require_buffer_by_literal::<#buffer>(#map_key, buffers);
```

Comment:
Is there a reason why we can't use `require_buffer_by_name`? I tried changing to that and the tests still passes.

Reply:
By name will do a heap allocation if an error occurs while by literal will not, since it can do a permanent borrow of the static memory. 

It's very minor but I figure we may as well take advantage of the savings.

Reply:
I feel that this is a very small optimization that makes the api confusing. There isn't any way to avoid passing a `&'static str` to `require_buffer_by_name`, the best we can do is take a `&String`, but imo that creates more downsides than it's worth. I guess the best can be done is to have more docs.